internal/reader/onchain_prices_reader.go

```diff
@@ -84,5 +94,67 @@ func (pr *OnchainTokenPricesReader) GetTokenPricesUSD(
 	return prices, nil
 }
 
+func (pr *OnchainTokenPricesReader) getFeedDecimals(ctx context.Context, token types.Account) (*uint8, error) {
+	var decimals *uint8
+	if err :=
+		pr.ContractReader.GetLatestValue(
+			ctx,
+			consts.ContractNamePriceAggregator,
+			consts.MethodNameGetDecimals,
+			primitives.Unconfirmed,
+			nil,
+			&decimals,
+			//boundContract,
+		); err != nil {
+		return nil, fmt.Errorf("decimals call failed for token %s: %w", token, err)
+	}
+
+	return decimals, nil
+}
+
+func (pr *OnchainTokenPricesReader) getRawTokenPriceE18Normalized(ctx context.Context, token types.Account) (*big.Int, error) {
+	var latestRoundData *LatestRoundData
+	if err :=
+		pr.ContractReader.GetLatestValue(
+			ctx,
+			consts.ContractNamePriceAggregator,
+			consts.MethodNameGetLatestRoundData,
+			primitives.Unconfirmed,
+			nil,
+			&latestRoundData,
```

Comment:
Is this correct? We're passing in a pointer to a pointer?

Reply:
Yes, it's passing integration tests.